File: apps/ios/ScribbleFit/ScribbleFit/Features/Canvas/Domain/Repository/WorkoutSessionRepository.swift
import Foundation

public protocol WorkoutSessionRepository: Sendable {
    /**
     * Observes the current active, uncommitted workout session.
     */
    func getActiveSession() async throws -> WorkoutSession?

    /**
     * Updates or creates an active session.
     */
    func updateSession(_ session: WorkoutSession) async throws

    /**
     * Clears the active session after it has been committed to the ledger.
     */
    func clearActiveSession() async throws
}

File: apps/android/feature/ai/domain/src/main/java/com/scribblefit/feature/ai/domain/repository/WorkoutSessionRepository.kt
package com.scribblefit.feature.ai.domain.repository

import com.scribblefit.feature.ai.domain.model.WorkoutSession
import kotlinx.coroutines.flow.Flow

interface WorkoutSessionRepository {
    /**
     * Observes the current active, uncommitted workout session.
     */
    fun getActiveSession(): Flow<WorkoutSession?>

    /**
     * Updates or creates an active session with new data from the AI parser.
     */
    suspend fun updateSession(session: WorkoutSession)

    /**
     * Clears the active session after it has been committed to the ledger.
     */
    suspend fun clearActiveSession()
}
